content/03-type-system.md

```diff
@@ -1,31 +1,31 @@
 <!--label:type-system-->
 ## Type System
 
-We learned about the different kinds of types in [Types](types) and it is now time to see how they interact with each other. We start off easy by introducing [typedef](type-system-typedef), a mechanism to give a name (or alias) to a more complex type. Among other things, this will come in handy when working with types having [type parameters](type-system-type-parameters).
+We learned about the different kinds of types in [Types](types) and it is now time to see how they interact with each other. We start off easy by introducing [typedef](type-system-typedef), a mechanism to give a name (or alias) to a more complex type. Among other use cases, typedefs will come in handy when working with types that have [type parameters](type-system-type-parameters).
 
-A lot of type-safety is achieved by checking if two given types of the type groups above are compatible. Meaning, the compiler tries to perform **unification** between them as detailed in [Unification](type-system-unification).
+ !!!-REVIEW ME: What is this sentence even referring to? "Type groups" is also unclear in the sense that this phrase has not been used before and may have special meaning of some sort-!!! A lot of type-safety is achieved by checking if two given types of the type groups above are compatible. Meaning, the compiler tries to perform **unification** between them as detailed in [Unification](type-system-unification).
```

Comment:
I think "of the type groups above" words can be removed. 
Looks like "type groups" means typedefs, abstracts, enums etc.

Reply:
Agreed, this can be removed.